test/http-multipart.js

```diff
@@ -0,0 +1,509 @@
+        spec: specOpenApi2,
+        operationId: 'post_land_content_ViewOfAuthOwner',
+        parameters: {
+          'formData.hhlContent: sort': 'id',
+          'formData.hhlContent: order': 'desc',
+          'formData.email[]': ["person1", "person2"] // eslint-disable-line quotes
+        }
+      })
+      expect(req).toMatchObject({
+        method: 'POST',
+        url: '/api/v1/land/content/ViewOfAuthOwner',
+        credentials: 'same-origin',
+        headers: {
+          'Content-Type': 'multipart/form-data'
+        },
+      })
+      const fdArrayItem = req.body.getAll('email[]')
+      expect(fdArrayItem.length).toEqual(2)
+      expect(fdArrayItem[0]).toEqual('person1')
+      expect(fdArrayItem[1]).toEqual('person2')
+    })
+    /**
+     * Dev test only: assumes local server exists for POST
+     * Expect server response format: { message: 'ok', data: returnData }
+     * where returnData = req.body/req.files/req.file
+     */
+    it.skip('should (Live) POST multipart-formdata with array text', () => {
+      // when
+      const req = buildRequest({
+        spec: specOpenApi2,
+        operationId: 'post_land_content_ViewOfAuthOwner',
+        parameters: {
+          'hhlContent: sort': 'id',
+          'hhlContent: order': 'desc',
+          'email[]': ["person1", "person2"] // eslint-disable-line quotes
+        }
+      })
+      expect(req).toMatchObject({
+        method: 'POST',
+        url: '/api/v1/land/content/ViewOfAuthOwner',
+        credentials: 'same-origin',
+        headers: {
+          'Content-Type': 'multipart/form-data'
+        },
+      })
+      const fdArrayItem = req.body.getAll('email[]')
+      expect(fdArrayItem.length).toEqual(2)
+      expect(fdArrayItem[0]).toEqual('person1')
+      expect(fdArrayItem[1]).toEqual('person2')
+      const validateFormDataInstance = req.body instanceof FormData
+      expect(validateFormDataInstance).toEqual(true)
+      return fetch('http://localhost:3300/api/v1/formdata', { // eslint-disable-line no-undef
+        method: 'POST',
+        body: req.body.stream, // per formdata-node docs
+        headers: req.body.headers // per formdata-node docs
+      })
+        .then((res) => {
+          return res.json()
+        })
+        .then((json) => {
+          expect(json.email.length).toEqual(2)
+          expect(json.email[0]).toEqual('person1')
+          expect(json.email[1]).toEqual('person2')
+        })
+        .catch((err) => {
+        })
+    })
+    it('should Mock POST multipart-formdata with array text', () => {
+      // Given
+      fetchMock.post('http://localhost:3300/api/v1/formdata', {
+        status: 200,
+        body: {
+          message: 'post received',
+          data: {
+            'hhlContent: sort': 'id',
+            'hhlContent: order': 'desc',
+            email: ['person1', 'person2']
+          }
+        }
+      },
+        {
+          sendAsJson: false
+        })
+      // when
+      const req = buildRequest({
```

Comment:
for both oas2/oas3, the tests are checking that the FormData entries are not changed. updating test description to `should return FormData entry list and entry item entries (in order)`

Reply:
the basic issue that this PR fixes is that the entries were being squashed into a single string.

Reply:
Ok let me demonstrate what I mean here. Tests should contains only couple of asserts, in order to keep them readable. If assertions reaches some level, tests needs to broken down into multiple tests that tests different aspect of the `subject`.

```js
describe('buildRequest - openapi 2.0', () => {
  describe('given request object with formData array field', () => {
    const request = buildRequest({
      spec: sampleMultipartOpenApi2,
      operationId: 'post_land_content_ViewOfAuthOwner',
      parameters: {
        'formData.hhlContent:sort': 'id',
        'formData.hhlContent:order': 'desc',
        'formData.email[]': ["person1", "person2"] // eslint-disable-line quotes
      }
    })

    test('should build request body as formData', () => {
      const itemEntries = request.body.getAll('email[]')
      expect(itemEntries.length).toEqual(2)
      expect(itemEntries[0]).toEqual('person1')
      expect(itemEntries[1]).toEqual('person2')
    });

    test('should return appropriate response media type', () => {
      expect(request).toMatchObject({
        method: 'POST',
        url: '/api/v1/land/content/ViewOfAuthOwner',
        credentials: 'same-origin',
        headers: {
          'Content-Type': 'multipart/form-data'
        },
      })
    })
  });
});
```

Jest can nest `describe` blocks to create logical DSL language similar to Gherkins. Then we have two separate tests testing only one aspect of the expectation instead of all in one place.

Reply:
I understand your point now. duplicative tests removed.

Reply:
Not sure I follow the `duplicate tests removed`. I was talking about structure of the tests, so that they're more readable.